libs/langchain/langchain/schema/runnable.py

```diff
@@ -192,6 +226,146 @@ def _call_with_config(
             )
             return output
 
+    def _transform_stream_with_config(
+        self,
+        input: Iterator[Input],
+        transformer: Callable[[Iterator[Input]], Iterator[Output]],
+        config: Optional[RunnableConfig],
+        run_type: Optional[str] = None,
+    ) -> Iterator[Output]:
+        from langchain.callbacks.manager import CallbackManager
+
+        # tee the input so we can iterate over it twice
+        input_for_tracing, input_for_transform = tee(input, 2)
```

Comment:
Because we cant consume the same iterator twice, we consume it for
- producing the output (input_for_transform)
- logging the combined input value to the tracer

Reply:
Sorry didn't delete this comment, this was from a first pass of reading

Reply:
If it helps, confusion caused because I was expecting the input to take an element rather than an iterable, and didn't realize that final input and final outputs were referring to accumulators on a first pass